### DISCUSSION TITLE
[Plugin Request] add support for typeorm template

### DISCUSSION THREAD

**Discussion by iamclaytonray** [dotansimha/graphql-code-generator#1254] [General]
Because GraphQL and TypeORM are both typed, it would be _fairly_ easy to add support to the generator for TypeORM.

Given this example schema:

```graphql
type User {
  id: ID!
  createdAt: DateTime!
  updatedAt: DateTime!
  email: String!
  password: String!
  posts: [Post]
}
```

the expected output would be:

```typescript

export class User extends BaseEntity {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @CreateDateColumn()
  createdAt: Date;

  @UpdateDateColumn()
  updatedAt: Date;

  @Column({ nullable: false })
  email: string;

  @Column({ nullable: false })
  password: string;

  @OneToMany(type => Post)
  posts?: Post[];
}
```

We would need a way to ignore certain object types, such as something like:

```graphql
type AuthPayload {
  user: User!
  token: String!
}
```

Obviously, that should not be a database model. I can think of several other edge-cases where we wouldn't want to generate an entity for. I haven't gone in-depth with this tool yet but maybe something like a comment that a parser could read and tell the core to ignore from generating? This functionality could already be baked in, for all I know.

**Comment by dotansimha** [comment]
Hi @iamclaytonray ! Thank you for reporting it. 
Creating a template for TypeORM should be very similar to what we did in `typescript-mongo`, but should reflect the TypeORM directives as GraphQL directives.

**Comment by ardatan** [comment]
But notice that we should warn users to disable `synchronize` in `ormconfig`. Otherwise, user can lose data in the database.

**Comment by lednhatkhanh** [comment]
Really excited to this happening
